tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,215 @@
+    pConnInfo->pAlpnProtocols = safeMalloc(pConnInfo->alpnProtocolsLen);
+    pConnInfo->pCaCert = malloc(sizeof(uint32_t));
+    pConnInfo->pClientCert = malloc(sizeof(uint32_t));
+    pConnInfo->pPrivateKey = malloc(sizeof(uint32_t));
+    pConnInfo->userBuffer.pBuffer = safeMalloc(sizeof(_connHandle_t));
+  }
+  return pConnInfo;
+}
+
+/* Creates a Connection Handle and assigns memory accordingly. */
+IotHttpsConnectionHandle_t newIotConnectionHandle () {
+  IotHttpsConnectionHandle_t pConnectionHandle = safeMalloc(sizeof(_connHandle_t));
+  if(pConnectionHandle) {
+    pConnectionHandle->pNetworkConnection = safeMalloc(sizeof(uint32_t));
+    pConnectionHandle->pNetworkInterface = newNetworkInterface();
+    pConnectionHandle->reqQ.pPrevious = NULL;
+    pConnectionHandle->reqQ.pNext = NULL;
+    pConnectionHandle->respQ.pPrevious = NULL;
+    pConnectionHandle->respQ.pNext = NULL;
+    //pConnectionHandle->taskPoolJob = malloc(sizeof(struct _taskPoolJob));
+  }
+  return pConnectionHandle;
+}
+
+/* Creates a Response Handle and assigns memory accordingly. */
+IotHttpsResponseHandle_t newIotResponseHandle() {
+  IotHttpsResponseHandle_t pResponseHandle = safeMalloc(sizeof(_resHandle_t));
+  if(pResponseHandle) {
+    pResponseHandle->pBody = malloc(sizeof(uint32_t));
+    pResponseHandle->pHttpsConnection = newIotConnectionHandle();
+    pResponseHandle->pHttpsRequest = safeMalloc(sizeof(_reqHandle_t));
+    if(pResponseHandle->pHttpsRequest) {
+      pResponseHandle->pHttpsRequest->pHttpsResponse = pResponseHandle;
+    }
+    pResponseHandle->pHeaders = safeMalloc(sizeof(pResponseHandle) + sizeof(struct _httpResponse));
+    pResponseHandle->pHeadersCur = pResponseHandle->pHeaders;
+    pResponseHandle->httpParserInfo.readHeaderParser.data = pResponseHandle;
+    pResponseHandle->httpParserInfo.parseFunc = http_parser_execute;
+    pResponseHandle->pReadHeaderValue = safeMalloc(sizeof(uint32_t));
+    pResponseHandle->pHeadersEnd = safeMalloc(sizeof(pResponseHandle) + sizeof(_resHandle_t)); ;
+    __CPROVER_assume(pResponseHandle->readHeaderValueLength >= 0 &&
+    pResponseHandle->readHeaderValueLength <= (pResponseHandle->pHeadersEnd - pResponseHandle->pHeaders));
+    pResponseHandle->pReadHeaderValue = malloc(pResponseHandle->readHeaderValueLength);
+  }
+  return pResponseHandle;
+}
+
+/* Creates a Request Handle and assigns memory accordingly. */
+IotHttpsRequestHandle_t newIotRequestHandle() {
+  IotHttpsRequestHandle_t pRequestHandle = safeMalloc(sizeof(_reqHandle_t));
+  if (pRequestHandle) {
+    uint32_t len;
+    pRequestHandle->pHttpsResponse = newIotResponseHandle();
+    pRequestHandle->pHttpsConnection = newIotConnectionHandle();
+    pRequestHandle->pBody = malloc(len);
+    pRequestHandle->pHeaders = safeMalloc(sizeof(pRequestHandle) + sizeof(struct _httpRequest));
+    pRequestHandle->pHeadersCur = pRequestHandle->pHeaders;
+    pRequestHandle->pHeadersEnd = safeMalloc(sizeof(pRequestHandle) + sizeof(_reqHandle_t));
```

Comment:
` pRequestHandle->pHeadersEnd = pRequestHandle->pHeaders + sizeof(((_reqHandle_t*)pRequestHandle)->data)

Reply:
I did this. Resolving this comment.

Reply:
I have refreshed a few times it doesn't appear so

Reply:
Yes, sorry. Deleted.